src/java.net.http/share/classes/java/net/http/HttpRequest.java

```diff
@@ -720,6 +719,34 @@ public static BodyPublisher ofFile(Path path) throws FileNotFoundException {
             return RequestPublishers.FilePublisher.create(path);
         }
 
+        /**
+         * {@return a request body publisher whose body is the {@code length}
+         * content bytes read from the provided file {@code channel} starting
+         * from the specified {@code offset}}
+         * <p>
+         * The {@linkplain FileChannel file channel} will be read using
+         * {@link FileChannel#read(ByteBuffer, long) FileChannel.read(ByteBuffer buffer, long position)},
+         * which does not modify the channel's position. Thus, the same file
+         * channel may be shared between several publishers passed to
+         * concurrent requests.
+         * <p>
+         * The file channel will not be closed upon completion. The caller is
+         * expected to manage the life cycle of the channel, and close it
+         * appropriately when not needed anymore.
+         *
+         * @param channel a file channel
+         * @param offset the offset of the first byte
+         * @param length the number of bytes to read from the file channel
+         *
+         * @throws IndexOutOfBoundsException if the specified byte range is
+         * found to be out of bounds compared with the size of the file
+         * referred by the channel
```

Comment:
So we accept length = 0 (OK though not very useful?) and we reject length < 0 (which is good). Maybe we should say:
```suggestion
         * @throws IndexOutOfBoundsException if the specified byte range is
         * found to be {@linkplain Objects.checkFromIndexSize(long, long, long) out of bounds}
         * compared with the size of the file referred by the channel
```
to clarify that.